sdk/typescript/src/cryptography/ed25519-keypair.ts

```diff
@@ -79,6 +81,49 @@ export class Ed25519Keypair implements Keypair {
     return new Ed25519Keypair(nacl.sign.keyPair.fromSeed(seed));
   }
 
+  /**
+   * Test Derive Path from path string
+   *
+   * @param path path string (`m/44'/784'/0'/0'/0'`)
+   */
+  static isValidPath = (path: string): boolean => {
+    if (
+      !new RegExp("^m\\/44+'\\/784+'\\/[0-9]+'\\/[0-9]+'\\/[0-9]+'+$").test(
+        path
+      )
+    ) {
+      return false;
+    }
+    return !path
+      .split('/')
+      .slice(1)
+      .map((val: string): string => val.replace("'", ''))
+      .some(isNaN as any);
+  };
+
+  /**
+   * Generate a keypair from mnemonics.
+   *
+   * @param path path string (`m/44'/784'/0'/0'/0'`)
+   * @param mnemonics: mnemonics is a word seed phrase
+   */
+  static fromDerivePath(path: string, mnemonics: string): Ed25519Keypair {
+    if (!Ed25519Keypair.isValidPath(path)) {
+      throw new Error('Invalid derivation path');
+    }
+
+    const normalizeMnemonics = mnemonics
+      .trim()
+      .split(/\s+/)
+      .map(part => part.toLowerCase())
+      .join(' ');
```

Comment:
worth validating the mnemonics too? 

```
However, there should be other checks in place, such as checking to make sure the user is inputting 12 words or more separated by a space. ie. phrase.trim().split(/\s+/g).length >= 12
```

Reply:
For users who used normally created mnemonics, the process of changing wallets becomes easier. For example, using the HW Ledger registered with the same mnemonics, it can be used immediately.

In the case of an abnormally created mnemonics user, when changing to another wallet, it is inconvenient to use it after moving all of his/her assets to the new address created in the wallet.

In the case of near, the web wallet uses mnemonics of 12 words, and hw ledger always uses 24 words, so there is confusion in use. Fortunately, Near has the concept of an account ID, so you only need to register with a new accesskey, so you can avoid the confusion of creating a new address for each wallet.

So, if you plan to develop HWLedger, I think this part will be very necessary.